web/src/message_events.ts

```diff
@@ -551,7 +551,15 @@ export function update_messages(events: UpdateMessageEvent[]): void {
                 }
 
                 compose_validate.warn_if_topic_resolved(true);
-                compose_validate.inform_if_topic_is_moved(orig_topic, old_stream_id);
+
+                const topic_is_renamed =
+                    topic_edited &&
+                    resolved_topic.unresolve_name(orig_topic) !==
+                        resolved_topic.unresolve_name(new_topic);
+
+                if (stream_changed || topic_is_renamed) {
```

Comment:
Should this logic be inside `inform_if_topic_is_moved` so that the move banner is hidden when needed. Like for a case below the move banner is not hidden and resolve topic banner is shown -
- Initially only stream is changed. So, the move banner is shown.
- And then keeping the stream same, topic is resolved. But the move banner is still visible along with resolved banner.

![Peek 2025-07-16 21-29](https://github.com/user-attachments/assets/516e3635-5b72-4134-9618-03cbc63ce510)

Reply:
Yeah, that seems better.